api/controllers/AuthController.js

```diff
@@ -28,6 +28,27 @@ module.exports = {
     passport.authenticate('reddit', {
       state: req.session.state,
       duration: 'permanent',
+      failureRedirect: '/login',
+      scope: 'identity'
+    })(req, res);
+  },
+
+  modAuth: function(req, res) {
+    req.session.state = crypto.randomBytes(32).toString('hex');
+    if (req.query.url) {
+      req.session.redirectUrl = req.query.url;
+    }
+    passport.authenticate('reddit', {
+      state: req.session.state,
+      duration: 'permanent',
+      failureRedirect: '/login',
+    })(req, res);
```

Comment:
Yes, this works. It fixes the issue from yesterday, and it also fixes that error message which appeared after mod authentication.